### PR TITLE
Add listening port customization and simple HTTP Basic authentication support

### PR DESCRIPTION
**Pull request by jqueuniet**

### CODE REVIEW

File: templates/reverseproxy.conf.j2 [hunk 1]
```diff
@@ -5,8 +5,8 @@
 
 {% if item.key == "default" %}
 server {
-    listen        80 default_server;
-    listen   [::]:80 default_server;
+    listen        {{ item.value.listen }} default_server;
+    listen   [::]:{{ item.value.listen }} default_server;
```
Comment (hispanico) [inline templates/reverseproxy.conf.j2]: Missing listen default value

File: tasks/main.yml [hunk 2]
```diff
@@ -10,6 +10,28 @@
...
 
+- name: Install python-passlib
+  apt:
+    name:
+        - "python-passlib"
+        - "python3-passlib"
+    state: present
+  tags:
+    - nginx-revproxy
+    - packages
```
Comment (hispanico) [inline tasks/main.yml]: Why do we need this packages?
As in the ansible docs, should't it need on the target host?

> The below requirements are needed on the host that executes this module.
> passlib>=1.6

File: tasks/main.yml [hunk 3]
```diff
@@ -10,6 +10,28 @@
...
+
+- name: Add authentication
+  htpasswd:
+    path: "/etc/nginx/{{ item.key }}_htpasswd"
+    name: "{{ item.value.auth.login }}"
+    password: "{{ item.value.auth.password }}"
+  loop: "{{ nginx_revproxy_sites|dict2items }}"
```
Comment (hispanico) [inline tasks/main.yml]: If we use loop with ```dict2items```, we need to change the Minimum Ansible Version to 2.6

**Comment by hispanico** [comment]
Good job.
See my inline comments.

File: tasks/main.yml [hunk 4]
```diff
@@ -10,6 +10,28 @@
     - nginx-revproxy
     - packages
 
+- name: Install python-passlib
+  apt:
+    name:
+        - "python-passlib"
```
Comment (hispanico) [inline tasks/main.yml]: Why do we need this packages?
As in the ansible docs, should't it need on the target host?

    The below requirements are needed on the host that executes this module.
    passlib>=1.6

**Comment by jqueuniet** [comment]
Changed the defaults and the loop.

The passlib package is a requirement for the htpasswd module, has to be installed on any host targeted by this playbook, or else you get errors like this:

```failed: [worker-000] (item={'key': 'PUBLICIP_node-exporter', 'value': {'domains': ['PUBLICIP'], 'upstreams': [{'backend_address': '127.0.0.1', 'backend_port': 9001}], 'listen': 9002, 'auth': {'login': 'container', 'password': '****************'}, 'ssl': False, 'letsencrypt': False}}) => {"changed": false, "item": {"key": "{{ ansible_host }}_node-exporter", "value": {"auth": {"login": "container", "password": "***********"}, "domains": ["PUBLICIP"], "letsencrypt": false, "listen": 9002, "ssl": false, "upstreams": [{"backend_address": "127.0.0.1", "backend_port": 9001}]}}, "msg": "This module requires the passlib Python library"}```

I changed the install task to only pull the package for the relevant Python version.

**Pull request merged**